diff --git a/src/c.zig b/src/c.zig
index e7d2905..eee8f0e 100644
--- a/src/c.zig
+++ b/src/c.zig
@@ -1,52 +1,8 @@
 const __root = @This();
 pub const __builtin = @import("std").zig.c_translation.builtins;
 pub const __helpers = @import("std").zig.c_translation.helpers;
-pub const TPixel = extern struct {
-    r: u8 = 0,
-    g: u8 = 0,
-    b: u8 = 0,
-    a: u8 = 0,
-};
-pub const struct_Tigr = extern struct {
-    w: c_int = 0,
-    h: c_int = 0,
-    cx: c_int = 0,
-    cy: c_int = 0,
-    cw: c_int = 0,
-    ch: c_int = 0,
-    pix: [*c]TPixel = null,
-    handle: ?*anyopaque = null,
-    blitMode: c_int = 0,
-    pub const tigrFree = __root.tigrFree;
-    pub const tigrClosed = __root.tigrClosed;
-    pub const tigrUpdate = __root.tigrUpdate;
-    pub const tigrBeginOpenGL = __root.tigrBeginOpenGL;
-    pub const tigrSetPostShader = __root.tigrSetPostShader;
-    pub const tigrSetPostFX = __root.tigrSetPostFX;
-    pub const tigrGet = __root.tigrGet;
-    pub const tigrPlot = __root.tigrPlot;
-    pub const tigrClear = __root.tigrClear;
-    pub const tigrFill = __root.tigrFill;
-    pub const tigrLine = __root.tigrLine;
-    pub const tigrRect = __root.tigrRect;
-    pub const tigrFillRect = __root.tigrFillRect;
-    pub const tigrCircle = __root.tigrCircle;
-    pub const tigrFillCircle = __root.tigrFillCircle;
-    pub const tigrClip = __root.tigrClip;
-    pub const tigrBlit = __root.tigrBlit;
-    pub const tigrBlitAlpha = __root.tigrBlitAlpha;
-    pub const tigrBlitTint = __root.tigrBlitTint;
-    pub const tigrBlitMode = __root.tigrBlitMode;
-    pub const tigrLoadFont = __root.tigrLoadFont;
-    pub const tigrPrint = __root.tigrPrint;
-    pub const tigrMouse = __root.tigrMouse;
-    pub const tigrTouch = __root.tigrTouch;
-    pub const tigrScrollWheel = __root.tigrScrollWheel;
-    pub const tigrKeyDown = __root.tigrKeyDown;
-    pub const tigrKeyHeld = __root.tigrKeyHeld;
-    pub const tigrReadChar = __root.tigrReadChar;
-    pub const tigrError = __root.tigrError;
-};
+pub const TPixel = @import("main.zig").Pixel;
+pub const struct_Tigr = @import("main.zig").Tigr;
 pub const Tigr = struct_Tigr;
 pub extern fn tigrWindow(w: c_int, h: c_int, title: [*c]const u8, flags: c_int) [*c]Tigr;
 pub extern fn tigrBitmap(w: c_int, h: c_int) [*c]Tigr;
